File: solutions/1280-diet-plan-performance/diet-plan-performance.py
class Solution:
    def dietPlanPerformance(self, calories: List[int], k: int, lower: int, upper: int) -> int:
        n = len(calories)
        left = 0
        right = 0
        running_sum = 0
        points = 0
        
        while right < n:
            running_sum += calories[right]
            
            if right - left + 1 == k:
                if running_sum < lower:
                    points -= 1
                    
                elif running_sum > upper:
                    points += 1
                
                running_sum -= calories[left]
                left += 1
            
            right += 1
            
        return points

File: solutions/1280-diet-plan-performance/diet-plan-performance.cpp
class Solution {
public:
    int dietPlanPerformance(vector<int>& calories, int k, int lower,
                            int upper) {
        int sum = 0, score = 0;
        for (int i = 0; i < k; i++) {
            sum += calories[i];
        }
        for (int i = k; i < calories.size(); i++) {
            if (sum < lower) {
                score--;
            } else if (sum > upper) {
                score++;
            }
            sum += calories[i] - calories[i - k];
        }
        if (sum < lower) {
            score--;
        } else if (sum > upper) {
            score++;
        }
        return score;
    }
};